EventListener/VersionListener.php

```diff
@@ -53,16 +53,24 @@ public function onKernelRequest(GetResponseEvent $event)
         }
 
         $this->version = $this->versionResolver->resolve($request);
+        if (false === $this->version && $request->attributes->has('version')) {
+            $this->version = $request->attributes->get('version');
+        } elseif (false === $this->version && null !== $this->defaultVersion) {
+            $this->version = $this->defaultVersion;
+        }
+
         if (false === $this->version && null !== $this->defaultVersion) {
             $this->version = $this->defaultVersion;
         }
 
+        // Override request attribute
         if (false !== $this->version) {
             $request->attributes->set('version', $this->version);
+        }
 
-            if ($this->viewHandler instanceof ConfigurableViewHandlerInterface) {
-                $this->viewHandler->setExclusionStrategyVersion($this->version);
-            }
+        // Use the resolved version when rendering the response
+        if ($request->attributes->has('version') && $this->viewHandler instanceof ConfigurableViewHandlerInterface) {
```

Comment:
We can now save a method call here by checking for `$this->version` instead of the request attributes.